### ISSUE TITLE
Support :repositories key in quoted project map

### ISSUE THREAD

**Issue by emlyn** [hyPiRion/inlein#7]
To allow use of dependencies that are not in the default repositories

**Comment by hyPiRion** [comment]
Yeah, I want to support this fairly quickly, as I need it myself.

However, for credentials-protected repos, I'd like to not insert credentials or how the credentials are resolved directly into the scripts. The repo itself is fine, but credentials and the like should probably reside in `~/.inlein/config.clj` or something similar.

**Comment by coreagile** [comment]
I also need this. We use private repositories to store our artifacts, and even worse, we use S3 to store our artifacts (using the s3-wagon plugin in leiningen). That would make this so useful for us!
